components/playground/main.go

```diff
@@ -266,6 +298,148 @@ Examples:
+		err = errors.Errorf("unknown playground mode: %s", modeVal)
+		return
+	}
+
+	if err = populateBoolOptIfChanged(flagSet, withMonitor, &options.Monitor); err != nil {
+		return
+	}
+
+	if err = populateIntOptIfChanged(flagSet, db, &options.TiDB.Num); err != nil {
+		return
+	}
+	if err = populateIntOptIfChanged(flagSet, kv, &options.TiKV.Num); err != nil {
+		return
+	}
+	if err = populateIntOptIfChanged(flagSet, pd, &options.PD.Num); err != nil {
+		return
+	}
+	if err = populateIntOptIfChanged(flagSet, tiflash, &options.TiFlash.Num); err != nil {
+		return
+	}
+	if err = populateIntOptIfChanged(flagSet, ticdc, &options.TiCDC.Num); err != nil {
+		return
+	}
+	if err = populateIntOptIfChanged(flagSet, pump, &options.Pump.Num); err != nil {
+		return
+	}
+	if err = populateIntOptIfChanged(flagSet, drainer, &options.Drainer.Num); err != nil {
+		return
+	}
+	if err = populateIntOptIfChanged(flagSet, dbTimeout, &options.TiDB.UpTimeout); err != nil {
+		return
+	}
+	if err = populateIntOptIfChanged(flagSet, tiflashTimeout, &options.TiFlash.UpTimeout); err != nil {
+		return
+	}
+
+	if err = populateStringOptIfChanged(flagSet, clusterHost, &options.Host); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, dbHost, &options.TiDB.Host); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, pdHost, &options.PD.Host); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, dbConfig, &options.TiDB.ConfigPath); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, kvConfig, &options.TiKV.ConfigPath); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, pdConfig, &options.PD.ConfigPath); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, tiflashConfig, &options.TiFlash.ConfigPath); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, ticdcConfig, &options.TiCDC.ConfigPath); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, pumpConfig, &options.Pump.ConfigPath); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, drainerConfig, &options.Drainer.ConfigPath); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, dbBinpath, &options.TiDB.BinPath); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, kvBinpath, &options.TiKV.BinPath); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, pdBinpath, &options.PD.BinPath); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, tiflashBinpath, &options.TiFlash.BinPath); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, ticdcBinpath, &options.TiCDC.BinPath); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, pumpBinpath, &options.Pump.BinPath); err != nil {
+		return
+	}
+	if err = populateStringOptIfChanged(flagSet, drainerBinpath, &options.Drainer.BinPath); err != nil {
+		return
+	}
+	return
```

Comment:
Can we use a loop to deal with this template code?

Reply:
how can we achieve that? the problem is keys(`ticdcBinpath`, `pdConfig`, ..) and containers(`&options.TiCDC.BinPath`, `&options.PD.ConfigPath`) all names. And there is no generic. So we may locally do the grouping and for a local lists, I don't think it is a much more clever way.

Reply:
Please see if the latest refactor as you want.